src/synapse/azext_synapse/custom.py

```diff
@@ -12,38 +13,59 @@
-    SqlPool, Sku
+    SqlPool, SqlPoolPatchInfo, Sku
 
+from .constant import DOTNET_CLASS, DOTNET_FILE, SPARK_DOTNET_UDFS_FOLDER_NAME, EXECUTOR_SIZE, \
+    SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY, SparkBatchLanguage, SynapseSqlCreateMode
 
-# pylint: disable=too-many-locals, too-many-branches, too-many-statements, unused-argument
+from ._client_factory import cf_synapse_client_workspace_factory
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements, unused-argument, too-many-function-args
 def list_spark_batch_jobs(cmd, client, workspace_name, spark_pool_name, from_index=None, size=None, detailed=True):
     return client.list(workspace_name, spark_pool_name, from_index, size, detailed)
 
 
-def create_spark_batch_job(cmd, client, workspace_name, spark_pool_name, job_name, args, driver_memory, driver_cores,
-                           executor_memory, executor_cores, num_executors, language="SCALA",
-                           file=None, class_name=None, jars=None, files=None,
-                           archives=None, conf=None,
+def create_spark_batch_job(cmd, client, workspace_name, spark_pool_name, job_name, main_definition_file,
+                           main_class_name, executor_size, executors, language=SparkBatchLanguage.Scala,
+                           command_line_arguments=None,
+                           reference_files=None, archives=None, configuration=None,
                            tags=None, detailed=True):
-    dotnet_file = "local:///usr/hdp/current/spark2-client/jars/microsoft-spark.jar"
-    dotnet_class = "org.apache.spark.deploy.dotnet.DotnetRunner"
-
-    if language.upper() != "DOTNET" and (not file or not class_name):
-        raise CLIError('Scala and Python spark batch job must provide value for parameter file and class_name.'
-                       'If you want to create a DotNet spark batch job please add `--language DOTNET`.')
-
-    if language.upper() == "DOTNET":
-        file = dotnet_file
-        class_name = dotnet_class
+    file = main_definition_file
+    class_name = main_class_name
+    args = command_line_arguments
+    conf = configuration
+    if language.upper() == SparkBatchLanguage.SparkDotNet.value.upper() or \
+            language.upper() == SparkBatchLanguage.CSharp.value.upper():
+        file = DOTNET_FILE
+        class_name = DOTNET_CLASS
+
+        args = [main_definition_file, main_class_name]
+        if command_line_arguments:
+            args = args + command_line_arguments
+
+        archives = ["{}#{}".format(main_definition_file, SPARK_DOTNET_UDFS_FOLDER_NAME)] + archives if archives \
+            else ["{}#{}".format(main_definition_file, SPARK_DOTNET_UDFS_FOLDER_NAME)]
+
+        if not conf:
+            conf = {SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY: './{}'.format(SPARK_DOTNET_UDFS_FOLDER_NAME)}
+        else:
+            conf[SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY] = './{}'.format(SPARK_DOTNET_UDFS_FOLDER_NAME)
+

```

Comment:
This logic might be incorrect. Can the SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY has multiple values?

Reply:
fixed.

---
In reply to: [417296842](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417296842) [](ancestors = 417296842)